### PR TITLE
PROD-877 Fixed-Not allowing apostrophes in nicknames

### LINKED ISSUE #1944: Not allowing apostrophes in nicknames
The filters for the names don't allow apostrophes which is causing issues with my communities' users as apostrophes are common in their names. (eg names like O'Brian, O'Reily, D'Andre, D'Armond, etc)

Can the bp_xprofile_validate_nickname_value function be changed to allow for apostrophes too?

The lines needing changing would be:

```
// only alpha numeric, underscore, dash
        if ( ! preg_match( '/^([A-Za-z0-9-_\.]+)$/', $value ) ) {
              return sprintf( __( 'Invalid %s. Only "a-z", "0-9", "-", "_" and "." are allowed.', 'buddyboss' ), $field_name );
        }
```
In the meantime I've been commenting out the lines, but this is is tedious to do every time. If it's not possible, or wanted, to add apostrophes to all systems, could this section to be changed so that folks could pick how/if they want to filter their nicknames in the settings? 

**Jira issue** : [PROD-877]

[PROD-877]: https://buddyboss.atlassian.net/browse/PROD-877?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### PR DESCRIPTION
**Pull request by meetmakadiya**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #1944 

### How to test the changes in this Pull Request:

1. Go to the user profile and edit the nickname.
2. use the apostrophes in nickname and try to save the value.

### Proof Screenshots or Video

http://i.imgur.com/vyyKJrn.png

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: src/bp-xprofile/bp-xprofile-filters.php [hunk 1]
```diff
@@ -785,8 +785,8 @@ function bp_xprofile_validate_nickname_value( $retval, $field_id, $value, $user_
 	}
 
 	// only alpha numeric, underscore, dash
-	if ( ! preg_match( '/^([A-Za-z0-9-_\.]+)$/', $value ) ) {
-		return sprintf( __( 'Invalid %s. Only "a-z", "0-9", "-", "_" and "." are allowed.', 'buddyboss' ), $field_name );
+	if ( ! preg_match( '/^([A-Za-z0-9\'-_\.]+)$/', $value ) ) {
+		return sprintf( __( 'Invalid %s. Only "a-z", "0-9", "\'", "-", "_" and "." are allowed.', 'buddyboss' ), $field_name );
```
Comment (dipskakadiya) [inline src/bp-xprofile/bp-xprofile-filters.php:789]: @meetmakadiya In the error message no need to add `\` with `'`.

**Review by dipskakadiya — CHANGES_REQUESTED** [review]
@meetmakadiya Check my comment.

File: src/bp-xprofile/bp-xprofile-filters.php (hunk 1 above)
Comment (meetmakadiya) [reply src/bp-xprofile/bp-xprofile-filters.php:789]: @dipskakadiya sprintf starting with` ' `so we need to add` \` without` \` like `" ' "` through error .
http://i.imgur.com/9lE8Fo9.png
http://i.imgur.com/16Hmg8M.png

Comment (dipskakadiya) [reply src/bp-xprofile/bp-xprofile-filters.php:789]: @meetmakadiya Thanks for detailed info. Looks good 👍

**Review by dipskakadiya — APPROVED** [review]

**Pull request closed without merging**